packages/next/src/build/flying-shuttle/store-shuttle.ts

```diff
@@ -8,20 +12,57 @@ import {
   PAGES_MANIFEST,
   ROUTES_MANIFEST,
 } from '../../shared/lib/constants'
-import { recursiveCopy } from '../../lib/recursive-copy'
+
+export function getShuttleManifest(config: NextConfigComplete) {
+  const globalHash = crypto.createHash('sha256')
+
+  for (const key of Object.keys(process.env || {})) {
+    if (key.startsWith('NEXT_PUBLIC_')) {
+      globalHash.update(`${key}=${process.env[key]}`)
+    }
+  }
+
+  const omittedConfigKeys = ['headers', 'rewrites', 'redirects']
+
+  for (const key of Object.keys(config)) {
```

Comment:
nit, I think this could be `for (const key in config)` without needing to allocate an array for all the keys in config